lib/rex/post/meterpreter/ui/console.rb

```diff
@@ -99,7 +99,13 @@ def queue_cmd(cmd)
   #
   def run_command(dispatcher, method, arguments)
     begin
-      super
+      super 
+    rescue Rex::TimeoutError => e
```

Comment:
Looks like the semantics here are bit off; the function is called `on_error_proc` - which is very generic, but you'll only call it in the the scenario of a `Rex::TimeoutError` exception being raised. I think this would break the https://en.wikipedia.org/wiki/Principle_of_least_astonishment

Reply:
To be more explicit; We should _always_ be calling `on_error_proc` - not just for the context that you would like to have handled in your current ticket